episodes/anonymization.md

```diff
@@ -32,19 +33,352 @@ Metadata elements in imaging, such as patient names and addresses, are often cle
 
+
+### Text on Images
+
+Occasionally, technicians will burn information directly onto images as part of a burned-in annotation. This may include details such as diagnoses, demographics, or the patient's name. Fortunately, this text is usually typed rather than handwritten, making it recognizable by optical character recognition (OCR) functions. Often, this text is placed away from the center of the image, allowing for clever cropping to eliminate it entirely in some datasets.
+
+
+
+::::::::::::::::::::::::::::::::::::::: challenge
+
+## Challenge: Getting rid of identifying burned in data
+
+An ultrasound (scraped from the public internet on a creative commons license lisence [here](https://www.flickr.com/photos/jcarter/2461223727).) must be kept at it's existing height and width dimensions. You should change the image from RGB to grayscale. It can be opened as follows:
+
+```python
+
+import numpy as np
+import matplotlib.pyplot as plt
+from skimage import io, filters
+
+image = 'data/anonym/identifiable_us.jpg'
+io.imshow(image)
+io.show()
+```
+
+```output
+```
+
+![Image from flikr website published with a permissive lisence.](fig/identifiable_us.jpg){alt='Identifiable ultrasound'}
+
+Write code for two approaches that de-identify (remove the annotations from) the ultrasound image data. 
+
+
+::::::::::::::: solution
+
+## Solution
+
+The image  has identifying metdata. You should identify not only the name, but the date and place as problematic.  You can take three different approaches. One would be to mask the data, the other would be to blur it. First we will show a blurred image:
+
+```python
+
+from skimage.filters import gaussian
+from skimage.color import rgb2gray
+
+image_base = io.imread(image)
+image_base = rgb2gray(image_base)
+sub_im = image_base[0:78,:].copy()
+blur_sub_im = gaussian(sub_im, sigma=9)
+final_image = np.zeros(image_base.shape)
+final_image[0:78,:] = blur_sub_im
+final_image[79:,:]= image_base[79:, :]
+io.imshow(final_image)
+
+```
+
+
+```output
+```
+
+![Image after blurring in one area.](fig/blurred_us.png){alt='Non-Identifiable blurred ultrasound'}
+We could have also just make a simple zero-mask: 
+
+```python
+image_masked = io.imread(image)
+image_masked[0:78,:] = 0
+io.imshow(image_masked)
+
+```
+
+
+```output
+```
+
+![Image after masking in one area.](fig/masked_us.png){alt='Non-Identifiable masked ultrasound'}
+
+Note there are other valid solutions, but these two are very common and straightforward. 
+
+Finally, you could always just chop off the offending part so to speak, and resize:
+
+```python
+from skimage.transform import resize
+
+final_image= image_base[79:, :]
+final_image = resize(final_image,image_base.shape)
```

Comment:
Should there be a warning here that this changes the aspect ratio of the image, thus distorting it?

Reply:
Agree. This must be highlighted in case there is a clueless copy-paster out there.